spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -985,75 +1313,123 @@ def write_data_spec(
-        if delayed_key is not None:
-            delayed_data, delayed_max_row_len = self._retrieve_synaptic_block(
-                transceiver, placement, master_pop_table, indirect_synapses,
-                direct_synapses, delayed_key,
-                pre_vertex_slice.n_atoms * app_edge.n_delay_stages,
-                index, using_extra_monitor_cores, placements,
-                monitor_api, extra_monitor, monitor_cores,
-                handle_time_out_configuration, fixed_routes=fixed_routes)
-
-        # Convert the blocks into connections
-        return self._read_synapses(
-            synapse_info, pre_vertex_slice, post_vertex_slice,
-            max_row_length, delayed_max_row_len, self.__n_synapse_types,
-            self.__weight_scales[placement], data, delayed_data,
-            machine_time_step)
+    def get_connections_from_machine(
+            self, vertex, transceiver, placements, app_edge, graph_mapper,
+            synapse_info, machine_time_step):
+
+        if not isinstance(app_edge, ProjectionApplicationEdge):
+            raise Exception("Unknown edge type for {}".format(app_edge))
+
+        post_vertices = graph_mapper.get_machine_vertices(vertex)
+        connections = []
+        progress = ProgressBar(
+            len(post_vertices),
+            "Getting synaptic data between {} and {}".format(
+                app_edge.pre_vertex.label, vertex.label))
+        for post_vertex in progress.over(post_vertices):
+            post_slice = graph_mapper.get_slice(post_vertex)
+            placement = placements.get_placement_of_vertex(post_vertex)
+            direct_synapses, indirect_synapses = \
+                self.__compute_addresses(transceiver, placement)
+
+            connections.extend(self._get_connections(
+                app_edge, synapse_info, post_slice,
+                transceiver, placement, direct_synapses, indirect_synapses,
+                machine_time_step, graph_mapper, self.__app_edge_info,
+                self.__m_edge_info, self.__synapse_io.read_undelayed_synapses))
+            connections.extend(self._get_connections(
+                app_edge, synapse_info, post_slice,
+                transceiver, placement, direct_synapses, indirect_synapses,
+                machine_time_step, graph_mapper,
+                self.__delay_edge_info, self.__delay_m_edge_info,
+                self.__synapse_io.read_delayed_synapses))
+        return numpy.concatenate(connections)
 
-    def __compute_addresses(self, transceiver, placement):
+    @staticmethod
+    def __compute_addresses(transceiver, placement):
         """ Helper for computing the addresses of the master pop table and\
             synaptic-matrix-related bits.
         """
-        master_pop_table = locate_memory_region_for_placement(
-            placement, POPULATION_BASED_REGIONS.POPULATION_TABLE.value,
-            transceiver)
         synaptic_matrix = locate_memory_region_for_placement(
```

Comment:
NO NO NO NO NO! we have to be able to support other vertices which are not APV which utilise a Synaptic manager. by forcing the regions to be that of APV is clunky. retrack to original format.